### PR TITLE
DEVEXP-615: SinchClient for mailgun

### PR DESCRIPTION
**Pull request by JPPortier**
- Support dedicated mailgun configuration
- Stub for authentication

Note: e2e tests failures are expected.
They are based onto `main` branch from `mockserver` but this feature branch do not contains the Voice related coverage

### CODE REVIEW

File: client/src/main/com/sinch/sdk/models/Configuration.java [hunk 1]
```diff
@@ -488,6 +523,63 @@ public Builder setConversationContext(ConversationContext context) {
...
 
+    /**
+     * Set Mailgun related api key
+     *
+     * @param apiKey {@link MailgunCredentials#getApiKey()}
+     * @return Current builder
+     * @since __TO_BE_DEFINED__
+     */
+    public Builder setMailgunApiKey(String apiKey) {
+      if (null == this.mailgunContext) {
```
Comment (asein-sinch) [inline client/src/main/com/sinch/sdk/models/Configuration.java]: check `this.mailgunCredentials` instead

Comment (JPPortier) [reply client/src/main/com/sinch/sdk/models/Configuration.java]: Right, seems a far better test ! 
Thank you.
Fixed

File: sample-app/src/main/java/com/sinch/sample/Utils.java [hunk 2]
```diff
@@ -21,6 +22,9 @@ public class Utils {
   private static final String SMS_SERVICE_PLAN_ID = "SMS_SERVICE_PLAN_ID";
   private static final String SMS_SERVICE_PLAN_TOKEN = "SMS_SERVICE_PLAN_TOKEN";
 
+  private static final String MAILGUN_API_KEY = "MAILGUN_SERVICE_API_KEY";
```
Comment (asein-sinch) [inline sample-app/src/main/java/com/sinch/sample/Utils.java:25]: Nit: Why would you name the properties differently?

Comment (JPPortier) [reply sample-app/src/main/java/com/sinch/sample/Utils.java:25]: From mailgun's dashboard, it is called "API key"
So, I  just mimic the term from dedicated dashboard
image

**Review by asein-sinch — APPROVED** [review]

**Pull request merged**